lib/Service/AiIntegrations/AiIntegrationsService.php

```diff
@@ -196,12 +196,25 @@ public function getSmartReply(Account $account, Mailbox $mailbox, Message $messa
 			} finally {
 				$client->logout();
 			}
-			$prompt = "Suggest 2 replies to the following email. Each reply should be 25 characters max. Separate the replies with  \"| \", like for example  \"Yes! | No, I'm not available \". Do not print anything else. The email contents are : ".$messageBody."";
+			$prompt = "You are tasked with formulating helpful replies or reply templates to e-mails provided that have been sent to me. If you don't know some relevant information for answering the e-mails (like my schedule) leave blanks in the text that can later be filled by me. You must write the replies from my point of view as replies to the original sender of the provided e-mail!
+
+			Formulate two extremely succinct reply suggestions to the provided ***E-MAIL***. Please, do not invent any context for the replies but, rather, leave blanks for me to fill in with relevant information where necessary. Provide the output formatted as valid JSON with the keys 'reply1' and 'reply2' for the reply suggestions.
+			
+			Each suggestion must be of 25 characters or less.
+
+			Here is the ***E-MAIL*** for which you must suggest the replies to:
+			
+			***START_OF_E-MAIL***".$messageBody."
+			
+			***END_OF_E-MAIL***
+			
+			Please, output *ONLY* a valid JSON string with the keys 'reply1' and 'reply2' for the reply suggestions. Leave out any other text besides the JSON! Be extremely succinct and write the replies from my point of view.
+			 ";
 			$task = new Task(FreePromptTaskType::class, $prompt, 'mail,', $currentUserId);
 			$manager->runTask($task);
-			$replies = array_slice(explode("|", $task->getOutput()), 0, 2);
-			$this->cache->addValue('smartReplies_'.$message->getUid(), implode("|", $replies));
-			return $replies;
+			$replies = $task->getOutput();
+			$this->cache->addValue('smartReplies_'.$message->getId(), $replies);
+			return json_decode($replies, true);
```

Comment:
open the code with phpstorm and let it fix the json_decode to throw on error ;-)

Reply:
and don't cache if you can't decode the replies. else the same decoding error happens again and again.